File: ehaugberg/01/01.js
const fs = require('fs');

const EMPTY = '';

const getSums = () => {
  const input = fs.readFileSync('input.txt', {encoding: 'utf-8'});
  const inputSplit = input.split('\n');

  const sums = inputSplit.reduce((acc, value) => {
    if (value === EMPTY) {
      return {currentSum: 0, sums: [acc.currentSum, ...acc.sums]};
    }

    return {currentSum: acc.currentSum + parseInt(value), sums: acc.sums};
  }, {currentSum: 0, sums: []})

  // only interested in the final list of sums
  return sums.sums;
}

const partOne = () => {
  return getSums().sort().reverse()[0];
}

const partTwo = () => {
  const sum = (a, b) => a + b;
  const desc = (a, b) => b - a;

  return getSums().sort(desc).slice(0, 3).reduce(sum);
}

console.log('part 1', partOne())
console.log('part 2', partTwo())

File: vililahtevanoja/day01/01.py
data = []
with open("input.txt") as f:
    data = [l.strip() for l in f.readlines()]

grouped_data = []
acc_data = []
for line in data:
    if line == "":
        grouped_data.append(acc_data)
        acc_data = []
        continue
    acc_data.append(int(line))
sums = [sum(ns) for ns in grouped_data]


print(f"Part 1: {max(sums)}")
print(f"Part 2: {sum(sorted(sums, reverse=True)[:3])}")